src/core/qgsfeature.cpp

```diff
@@ -127,6 +127,17 @@ QgsAttributes QgsFeature::attributes() const
   return d->attributes;
 }
 
+QVariantMap QgsFeature::attributeMap() const
+{
+  QVariantMap res;
+  const int fieldSize = d->fields.size();
+  for ( int i = 0; i < d->attributes.size() && i < fieldSize; ++i )
```

Comment:
What do you think about a sanity check here, if fieldSize != attributes.size() return an empty map and complain on all possible channels?

Reply:
I'll make it raise a ValueException!

Reply:
Nice!
A check here for c++ use would still be worth it?

Reply:
Yeah I'll add a debug msg for that.

Any objections to me backporting this to 3.22?

Reply:
Works for me